### PR TITLE
EPMEDU-1579 - Wrong feeding point icon colour and (point) state on the map on the home page after auto cancellation of the feeding booking

### PR DESCRIPTION
**Pull request by Andrii-Chornyi**
Implementation of task EPMEDU-1579.
-Cancelling feeding point reservation upon timer expiration:

https://user-images.githubusercontent.com/112079677/228340558-e35bf980-d736-4ed4-8aed-b6e0fa244c5e.mp4

### CODE REVIEW

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenEvent.kt [hunk 1]
```diff
@@ -15,6 +15,7 @@ sealed interface HomeScreenEvent {
         object Start : FeedingEvent
         object Cancel : FeedingEvent
         object Finish : FeedingEvent
+        data class Expired(val rejectMessage: String) : FeedingEvent
```
Comment (Fileek) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenEvent.kt]: Let's leave this event as an object and put the `rejectMessage` inside as a constant. 
I guess we don't need localisation for such case.

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenUI.kt [hunk 2]
```diff
@@ -72,20 +72,20 @@ internal fun HomeScreenUI(
         }
     }
 
+    LaunchedEffect(key1 = state.timerState) {
```
Comment (Fileek) [inline feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenUI.kt:75]: Can you explain why did you create a separate `LaunchedEffect`?

**Review by Fileek — CHANGES_REQUESTED** [review]

File: feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenUI.kt (hunk 2 above)
Comment (Andrii-Chornyi) [reply feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenUI.kt:75]: Existing launched effect uses key1 = state.timerState, key2 = state.feedingRouteState as keys. We change state of feedingRouteState by stopping it, so the whole process of rejecting expired route repeats and we can see an error as api call for rejecting route gets called second time, when it's already rejected

Comment (Fileek) [reply feature/home/src/main/java/com/epmedu/animeal/home/presentation/HomeScreenUI.kt:75]: Ah, I see, ok

**Review by Fileek — APPROVED** [review]

**Pull request merged**